rfcs/text/0014_scheduled_reports_new_plugin.md

```diff
@@ -0,0 +1,164 @@
+- No formal API on how to make pages themselves "exportable."
+- A convoluted REST API that doesn't lend itself well to being re-usable.
+- Requires at least two teams in order to integrate new pages.
+
+With the desire to build more sophisticated features, such as scheduled reports, the time has come to rethink how reporting operates so that page-specific behaviors belong to their respective owners. We see ownership start to become more concrete once we make the following statements:
+
+- Rendering characteristics of a page (Dashboards, for instance) should be maintained and owned by that team.
+- The general pipeline of reporting: queueing, export execution, browser binaries, persistance of assets, and the REST APIs are owned by the reporting team.
+- Other characteristics, such as licensing and role's, can be established in one (or both) places.
+
+At its core, we want to expand upon the features and fixes that can make reporting a killer-feature of Kibana. In order to do so, it's time to shed the maintenance cost and upkeep of application-specific logic to the underlying team's themselves.
+
+# Overall design
+
+The new plugin would follow Kibana's recommended and best practices: establishing a TypeScript-based plugin that others can add as a dependency and consume. The following is a brief overview of how an application would incorporate reporting into their application stack.
+
+## 1. Add a dependency on reporting
+In order to use reporting, a plugin simply has to add a dependency on reporting in their `kibana.json`:
+
+```json
+{
+  "id": "super-dashboards",
+  "version": "8.0.0",
+  "kibanaVersion": "kibana",
+  "optionalPlugins": ["reporting"],
+  "server": true,
+  "ui": true
+}
+```
+
+Now that `reporting` is added as a plugin, downstream plugins can now integrate reporting into their stack.
+
+## 2. Using the plugin
+
+Similar to other plugins, reporting will offer a module that allows consumers to define the types and kinds of reports allowed by the page. For example, the below call will register a new "PDF" report to the page:
+
+```js
+// plugin.ts server
+public setup(core: CoreSetup, plugins: PluginSetup) {
+  const { reporting } = plugins;
+
+  reporting.registerReport({
+    id: Symbol('super-dashboard-report'),
+    tags: ['super-dashboard', 'triple-word-score', 'synergy'],
+    type: 'pdf',
+    label: 'Export Dashboard',
+    visibility: 'user',
+    intervals: ['immediate', 'monthly', 'weekly'],
+    connectors: ['email', 'slack'],
+    before: (req, res, context): Promise<boolean> => {},
+    after: (result: Buffer) => void,
+  });
+}
+
+// plugin.ts public
+public start(core: CoreStart, plugins: PluginStart) {
+  const { reporting } = plugins;
+  const dashboardReports = reporting.findReportById(Symbol('super-dashboard-report'));
+}
+
+// Later, in some React component:
+<Button onClick={() => dashboardReports.open(`http://localhost:5601/my-dope-dashboard?pdf-export=true`)}>
```

Comment:
I'm thinking that `dashboardReports` is responsible for abstracting how the `JobParamsPDF` interface is set up, and interacting with the Report generation endpoint.

It looks like there is an implication that when the user wants to customize the schedule or actions, it's just internal state in `dashboardReports`, and there would be other methods for that.

Reply:
Nevermind the previous comment. Based on https://github.com/elastic/kibana/pull/80767/files#diff-8f69f8b20a7f69e71feac116a1b67d71ecc6770a3a79adf889869276c40f1b34R117 it looks like the `open` method is for showing the Reporting panel (currently `ScreenCapturePanelContent`)